### DISCUSSION TITLE
In the Go Starter Repo, use the official rules types

### DISCUSSION THREAD

**Discussion by JonathanArns** [BattlesnakeOfficial/feedback#105] [General]
I think it might be nice to, in the Go starter repo, import some of the types from the rules package instead of basically reimplementing them. Of course directly importing the types from rules would be a problem, because they don't have json struct tags, but if that could be changed, then I think it would be a neat improvement.

Or if not this, then at least name things the same and use the same integer types etc.

I found this a pretty big annoyance, since I based my snake on this starter repo and then at a later point wanted to
use some of the rules logic.

**Comment by bvanvugt** [comment]
It's a neat idea, curious what @robbles thinks... I'm also worried about how those two definitions may fall out of sync with upcoming projects.

**Comment by robbles** [comment]
@JonathanArns Are you thinking more of using the types that represent the structure of the `/move` requests? i.e.
https://github.com/BattlesnakeOfficial/rules/blob/5172c27eb0206af8266d34eb04a42925bf9e2e10/cli/commands/play.go?plain=1#L32-L93

or the internal types used in the rulesets, to represent the data after the request has been parsed?
